llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -4582,8 +4582,12 @@ BoUpSLP::LoadsState BoUpSLP::canVectorizeLoads(
     return LoadsState::Gather;
   }
 
+  // If ScalarTy is a FixedVectorType (when REVEC is enabled), it is hard for
+  // being strided load.
+  bool IsSourceScalarInstruction = !isa<FixedVectorType>(VL[0]->getType());
```

Comment:
!isa<VectorType>(VL[0]->getType());

Reply:
Why do we use `VectorType` here? SLP does not accept `ScalableVectorType`.

Reply:
That's exactly the point. This check will pass, if the type of VL[0] is scalable vector